mbed-util/CriticalSectionLock.h

```diff
@@ -47,8 +47,7 @@ class CriticalSectionLock {
 #ifdef TARGET_NORDIC
         sd_nvic_critical_region_enter(&_state);
 #else
-        _state = __get_PRIMASK();
-        __disable_irq();
+        _state = minar::platform::pushDisableIRQState();
```

Comment:
This is a bit strange. It calls a function from `minar::platform`, even if the module doesn't even depend on `minar-platform` directly. Also, `pushDisableIRQState` is something that should probably belong in `core-util`, not in `minar-platform`, since it's not really specific to `minar-platform`. All in all, it looks to me like the previous implementation made more sense. I'm not sure how this change helps you, but maybe you can rewrite it in a different way?